python-backend/app/api/documents/expected/resources/expected_document_uploads.py

```diff
@@ -80,7 +80,7 @@ def put(self, expected_document_guid):
             if not mine_doc:
                 return self.create_error_payload(404, 'Mine Document not found'), 404
 
-            expected_document.mine_documents.append(mine_doc)
+            expected_document.related_documents.append(mine_doc)
```

Comment:
Not sure why expected_document is being updated in this PR, but the line(s) below should be updated to use `save()` as per a recent chat with @Jsyro

Reply:
A lot of these random changes are due to the fact that what the frontend is expecting is named differently than what is on the models. I had to change around a lot of references in other classes due to this.